service/history/handler/handler_test.go

```diff
@@ -764,6 +764,464 @@ func (s *handlerSuite) TestRespondActivityTaskFailed() {
+		expectedError bool
+	}{
+		"valid input": {
+			caseName: "valid input",
+			input: &types.HistoryRespondActivityTaskCanceledRequest{
+				DomainUUID: testDomainID,
+				CancelRequest: &types.RespondActivityTaskCanceledRequest{
+					TaskToken: []byte("task-token"),
+					Details:   []byte("Details"),
+					Identity:  "identity",
+				},
+			},
+			expectedError: false,
+		},
+		"empty domainID": {
+			caseName: "empty domainID",
+			input: &types.HistoryRespondActivityTaskCanceledRequest{
+				DomainUUID: "",
+			},
+			expectedError: true,
+		},
+		"ratelimit exceeded": {
+			caseName: "ratelimit exceeded",
+			input: &types.HistoryRespondActivityTaskCanceledRequest{
+				DomainUUID: testDomainID,
+				CancelRequest: &types.RespondActivityTaskCanceledRequest{
+					TaskToken: []byte("task-token"),
+					Details:   []byte("Details"),
+					Identity:  "identity",
+				},
+			},
+			expectedError: true,
+		},
+		"token deserialization error": {
+			caseName: "token deserialization error",
+			input: &types.HistoryRespondActivityTaskCanceledRequest{
+				DomainUUID: testDomainID,
+				CancelRequest: &types.RespondActivityTaskCanceledRequest{
+					TaskToken: []byte("task-token"),
+					Details:   []byte("Details"),
+					Identity:  "identity",
+				},
+			},
+			expectedError: true,
+		},
+		"invalid task token": {
+			caseName: "invalid task token",
+			input: &types.HistoryRespondActivityTaskCanceledRequest{
+				DomainUUID: testDomainID,
+				CancelRequest: &types.RespondActivityTaskCanceledRequest{
+					TaskToken: []byte("task-token"),
+					Details:   []byte("Details"),
+					Identity:  "identity",
+				},
+			},
+			expectedError: true,
+		},
+		"get engine error": {
+			caseName: "get engine error",
+			input: &types.HistoryRespondActivityTaskCanceledRequest{
+				DomainUUID: testDomainID,
+				CancelRequest: &types.RespondActivityTaskCanceledRequest{
+					TaskToken: []byte("task-token"),
+					Details:   []byte("Details"),
+					Identity:  "identity",
+				},
+			},
+			expectedError: true,
+		},
+		"engine error": {
+			caseName: "engine error",
+			input: &types.HistoryRespondActivityTaskCanceledRequest{
+				DomainUUID: testDomainID,
+				CancelRequest: &types.RespondActivityTaskCanceledRequest{
+					TaskToken: []byte("task-token"),
+					Details:   []byte("Details"),
+					Identity:  "identity",
+				},
+			},
+			expectedError: true,
+		},
+	}
+
+	for name, input := range testInput {
+		s.Run(name, func() {
+			switch input.caseName {
+			case "valid input":
+				s.mockTokenSerializer.EXPECT().Deserialize(gomock.Any()).Return(&common.TaskToken{
+					WorkflowID: testWorkflowID,
+					RunID:      testValidUUID,
+				}, nil).Times(1)
+				s.mockShardController.EXPECT().GetEngine(testWorkflowID).Return(s.mockEngine, nil).Times(1)
+				s.mockEngine.EXPECT().RespondActivityTaskCanceled(gomock.Any(), input.input).Return(nil).Times(1)
+				s.mockRatelimiter.EXPECT().Allow().Return(true).Times(1)
```

Comment:
Having such dependency on test case name is not a pattern we would like to follow. Ideally each test case definition is self contained and doesn't require test body to do special things for it. 
You can expose a `mockFn(<mock objects as parameter>` on each test case and just call it here

Reply:
Thanks, I have applied those changes accordingly.